core/src/commonMain/kotlin/com/ustadmobile/core/viewmodel/report/detail/ReportDetailViewModel.kt

```diff
@@ -45,53 +61,106 @@ class ReportDetailViewModel(
         _appUiState.update { prev ->
             prev.copy(
                 fabState =
-                FabUiState(
-                    visible = true,
-                    text = systemImpl.getString(MR.strings.edit),
-                    icon = FabUiState.FabIcon.EDIT,
-                    onClick = this@ReportDetailViewModel::onClickEdit
-                )
+                    FabUiState(
+                        visible = true,
+                        text = systemImpl.getString(MR.strings.edit),
+                        icon = FabUiState.FabIcon.EDIT,
+                        onClick = this@ReportDetailViewModel::onClickEdit
+                    ),
+                title = "Graph",
             )
         }
+        viewModelScope.launch {
+            _uiState.whenSubscribed {
+                launchIfHasPermission(
+                    setLoadingState = true,
+                    permissionCheck = { true }
+                ) {
+                    val siteFlow = activeRepo.reportDao().findByUidLive(reportUid)
+                    launch {
+                        siteFlow.collect { report ->
```

Comment:
Needs to use collectLatest here such that if a new value is emitted, the function should cancel
Should also use distinctBy (with last modified time) to prevent a situation where if the same value is emitted multiple times it would be interrupted.

e.g. should be distinctBy (to filter out duplicates that didn't really change) then collectLatest